### PR TITLE
GEN-1645 - feat(ProductReviews): add truspilot tab

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Adds tab for _Trustpilot_ reviews; When clicked we'll shown _Trustpilot Widget_. _Trustpilot_ reviews will be enabled down in the stack.

image

## Justify why they are needed

First step to get _Trustpilot_ reviews into _ProductReviews_ feature.

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #3789** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3789?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #3790** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3790?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3791** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3791?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3789?utm_source=stack-comment).

File: apps/store/src/components/ProductReviews/ProductReviews.tsx [hunk 1]
```diff
@@ -1,104 +1,177 @@
 import styled from '@emotion/styled'
 import { useTranslation } from 'next-i18next'
-import { useState, useMemo } from 'react'
+import React, { useState } from 'react'
```
Comment (robinandeer) [inline apps/store/src/components/ProductReviews/ProductReviews.tsx:3]: not needed, prefer importing individual utilities :)

File: apps/store/src/components/ProductReviews/ProductReviews.tsx [hunk 2]
```diff
@@ -1,104 +1,177 @@
...
+
         <div>
-          {reviewsDistribution.map(([score, percentage]) => (
-            <ReviewsDistributionByScore key={score} score={score} percentage={percentage} />
-          ))}
+          {selectedTab === Tabs.PRODUCT &&
+            reviewsDistribution.map(([score, percentage]) => (
+              <ReviewsDistributionByScore key={score} score={score} percentage={percentage} />
+            ))}
+          {selectedTab === Tabs.TRUSTPILOT && <div>Trustpilot</div>}
```
Comment (robinandeer) [inline apps/store/src/components/ProductReviews/ProductReviews.tsx:84]: nit: add a todo comment here

File: apps/store/src/components/ProductReviews/ProductReviews.tsx [hunk 3]
```diff
@@ -1,104 +1,177 @@
...
+    console.warn('[ProductReviews]: No review data available. Skip rendering.')
+    return null
+  }
+  const { rating, comments, reviewsDistribution } = reviewsData
 
   return (
     <Wrapper y={3.5}>
       <Rating
-        score={averageRating.score}
+        score={Number(rating.score)}
```
Comment (robinandeer) [inline apps/store/src/components/ProductReviews/ProductReviews.tsx:53]: why is this needed? I thought the z schema would take care of it, no?

**Comment by guilhermespopolin** [comment]
> looks good overall :)
> 
> my general comment would be that "product page context" is becoming a problem though -- let's think about what to do with that in the future

Yes. I'll be working on moving that data outside Product Page context.

File: apps/store/src/components/ProductReviews/ProductReviews.tsx (hunk 3 above)
Comment (guilhermespopolin) [reply apps/store/src/components/ProductReviews/ProductReviews.tsx:53]: You're right. That conversion was forgotten here. Removing it.

**Review by gustaveen — APPROVED** [review]
👍

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Jan 2, 8:13 AM**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3789).
